llvm/lib/Transforms/Utils/SimplifyLibCalls.cpp

```diff
@@ -397,9 +397,11 @@ Value *LibCallSimplifier::emitStrLenMemCpy(Value *Src, Value *Dst, uint64_t Len,
 
   // We have enough information to now generate the memcpy call to do the
   // concatenation for us.  Make a memcpy to copy the nul byte with align = 1.
-  B.CreateMemCpy(
-      CpyDst, Align(1), Src, Align(1),
-      ConstantInt::get(DL.getIntPtrType(Src->getContext()), Len + 1));
+  B.CreateMemCpy(CpyDst, Align(1), Src, Align(1),
+                 ConstantInt::get(
+                     DL.getIntPtrType(Src->getContext(),
+                                      Src->getType()->getPointerAddressSpace()),
+                     Len + 1));
```

Comment:
This wrapping is making my eyes bleed. Use a temp variable for getIntPtrType?

Reply:
Plus there should be a CreateMemCpy overload that uses a constant size and figures out the type for you

Reply:
Adopted the form that figures out the size for you throughout.